File: backend/app/models/transcript_chunk.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Float, ForeignKey, Integer, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.database import Base


class TranscriptChunk(Base):

    __tablename__ = "transcript_chunks"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True
    )

    transcript_id: Mapped[int] = mapped_column(
        ForeignKey("transcripts.id"),
        nullable=False
    )

    chunk_index: Mapped[int] = mapped_column(
        Integer,
        nullable=False
    )

    start_time: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    end_time: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    text: Mapped[str] = mapped_column(
        Text,
        nullable=False
    )

    embedding_created: Mapped[bool] = mapped_column(
        Boolean,
        default=False
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow
    )

File: backend/Models/TranscriptChunk.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VideoIntelligencePlatform.Backend.Models;

[Table("transcript_chunks")]
public class TranscriptChunk
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("transcript_id")]
    public int TranscriptId { get; set; }

    [Column("chunk_index")]
    public int ChunkIndex { get; set; }

    [Column("start_time")]
    public double StartTime { get; set; }

    [Column("end_time")]
    public double EndTime { get; set; }

    [Column("text")]
    public string Text { get; set; } = string.Empty;

    [Column("embedding_created")]
    public bool EmbeddingCreated { get; set; } = false;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey("TranscriptId")]
    public Transcript? Transcript { get; set; }
}
